File: backend/internal/domain/models/logging.go
package domainmodels

type LogLevel string

const (
	LogLevelError LogLevel = "error"
	LogLevelWarn  LogLevel = "warn"
	LogLevelInfo  LogLevel = "info"
	LogLevelDebug LogLevel = "debug"
)

func (l LogLevel) IsValid() bool {
	switch l {
	case
		LogLevelError,
		LogLevelWarn,
		LogLevelInfo,
		LogLevelDebug:
		return true
	}
	return false
}

func (l LogLevel) Order() int {
	switch l {
	case LogLevelError:
		return 0
	case LogLevelWarn:
		return 1
	case LogLevelInfo:
		return 2
	case LogLevelDebug:
		return 3
	}
	return -1
}

type LogMeta map[string]any

File: backend/src/ips_app_old/domain/models/logging.py
from enum import IntEnum

class LogLevel(IntEnum):
    NONE = 0
    ERROR = 1
    WARN = 2
    INFO = 3
    DEBUG = 4
    @property
    def label(self) -> str:
        log_level_str = {
            LogLevel.NONE: "NONE",
            LogLevel.ERROR: "ERROR",
            LogLevel.WARN: "WARN",
            LogLevel.INFO: "INFO",
            LogLevel.DEBUG: "DEBUG"
        }
        return log_level_str[self]